### PR TITLE
[NativeAOT-LLVM] Delete the `IsRhpUnmanagedIndirection` workaround

### PR DESCRIPTION
**Pull request by SingleAccretion**
Instead, handle the CC difference at the point of use by routing the indirect call through a runtime import.

This fixes the dynamic generics test modulo some things which rely on threading and finalization (which is why I am not enabling it here).

### CODE REVIEW

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/RuntimeImports.cs [hunk 1]
```diff
@@ -972,6 +972,13 @@ public GCFrameRegistration(void* allocation, uint elemCount, bool areByRefs = tr
         internal static unsafe partial void RhCpuIdEx(int* cpuInfo, int functionId, int subFunctionId);
 #endif
 
+#if TARGET_WASM
+        [Intrinsic]
```
Comment (SingleAccretion) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/RuntimeImports.cs:976]:
```suggestion
```

**Comment by SingleAccretion** [comment]
@dotnet/nativeaot-llvm

File: src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.Llvm.cs [hunk 2]
```diff
@@ -83,7 +83,7 @@ public static int getSignatureForMethodSymbol(IntPtr thisHandle, IntPtr symbolHa
             if (node is IMethodNode { Offset: 0, Method: MethodDesc method })
             {
                 _this.Get_CORINFO_SIG_INFO(method, pSig, scope: null);
-                if (method.IsUnmanagedCallersOnly)
+                if (method.IsUnmanagedCallersOnly || node is RuntimeImportMethodNode)
```
Comment (jkotas) [inline src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.Llvm.cs]: `CORINFO_CALLCONV_UNMANAGED` implies GC transition. Are we compensating for it somewhere?

Comment (SingleAccretion) [reply src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.Llvm.cs]: > Are we compensating for it somewhere?

There is no need as the API is only used to retrieve the signature from a function pointer representative to construct the LLVM equivalent. Making it work for other cases would require adding a way to describe the "FCall calling convention", i. e. native ABI + no marshaling, no transition and no poll.

(Additionally, whether the transition will be needed is influenced by SPGCT and that is callsite specific.)

**Review by yowl — APPROVED** [review]

**Pull request merged**